segments/dir/dir.p9k

```diff
@@ -147,22 +147,32 @@ prompt_dir() {
         # to the length of the package path's string
         subdirectory_path=$(__p9k_truncate_path "${current_path:${#${(S%%)package_path//$~zero/}}}" $P9K_DIR_SHORTEN_LENGTH $P9K_DIR_SHORTEN_DELIMITER)
 
-        local pkgFile="unknown"
+        local pkgFile
         for file in "${P9K_DIR_PACKAGE_FILES[@]}"; do
           if [[ -f "${package_path}/${file}" ]]; then
             pkgFile="${package_path}/${file}"
             break;
           fi
         done
 
-        local packageName=$(jq '.name' ${pkgFile} 2> /dev/null \
-          || node -e 'console.log(require(process.argv[1]).name);' ${pkgFile} 2>/dev/null \
-          || cat "${pkgFile}" 2> /dev/null | grep -m 1 "\"name\"" | awk -F ':' '{print $2}' | awk -F '"' '{print $2}' 2>/dev/null \
-          )
-        if [[ -n "${packageName}" ]]; then
-          # Instead of printing out the full path, print out the name of the package
-          # from the package.json and append the current subdirectory
-          current_path="`echo ${packageName} | tr -d '"'`$subdirectory_path"
+        if [ -f "${pkgFile}" ]; then
+          local packageName=$(jq '.name' ${pkgFile} 2> /dev/null \
+            || node -e 'console.log(require(process.argv[1]).name);' ${pkgFile} 2>/dev/null \
+            || cat "${pkgFile}" 2> /dev/null | grep -m 1 "\"name\"" | awk -F ':' '{print $2}' | awk -F '"' '{print $2}' 2>/dev/null \
+            )
+          if [[ -n "${packageName}" ]]; then
+            # Instead of printing out the full path, print out the name of the package
+            # from the package.json and append the current subdirectory
+            current_path="`echo ${packageName} | tr -d '"'`$subdirectory_path"
+          fi
+        elif [ -n "${package_path}" ] && [ -n "${P9K_DIR_SHORTEN_GIT_FALLBACK}" ]; then
+          local repo_name
+          if [ -n "${P9K_DIR_SHORTEN_GIT_FALLBACK_USE_REMOTE}" ]; then
+            repo_name="${${$(git config --get remote."${P9K_DIR_SHORTEN_GIT_FALLBACK_REMOTE_NAME-origin}".url):t}%.git}"
```

Comment:
I don't see how `P9K_DIR_SHORTEN_GIT_FALLBACK_REMOTE_NAME` would get much use since changing it would change it globally and there are probably very little people who just have repositories with a remote that is other than `origin`. Another idea could be to find out some remote with `${$(command git remote)[1]}`.

Also, I'd use `command` for external commands to avoid hitting an alias of function.

Reply:
I was trying to keep external commands to a minimum, which is why I opted against automatic discovery of the remote. I think your point about `P9K_DIR_SHORTEN_GIT_FALLBACK_REMOTE_NAME` is valid, but is there any harm in leaving it there?

Agree'd re: `command`, I'll fix that.

Reply:
> Also, I'd use `command` for external commands to avoid hitting an alias of function.

Fixed

Reply:
A little thought experiment: You get a repo without `(package|composer).json` without an origin remote. Will you use this option? My opinion is, you won't because it will break all other repos and just rename your remote.